site/cds_rdm/legacy/resolver.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2024 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Resolver."""
+
+from flask import g
+from invenio_pidstore.models import PersistentIdentifier
+from invenio_rdm_records.proxies import current_rdm_records_service
+from invenio_search.engine import dsl
+
+from .errors import VersionNotFound
+
+
+def get_pid_by_legacy_recid(legacy_recid):
+    """Get record by pid value."""
+    # Get the object uuid from pidstore
+    recid = PersistentIdentifier.query.filter_by(
+        pid_value=legacy_recid, pid_type="lrecid"
+    ).one()
+
+    # Use the object uuid to get the parent pid value
+    parent_pid = PersistentIdentifier.query.filter_by(
+        object_uuid=recid.object_uuid, pid_type="recid"
```

Comment:
```suggestion
        object_uuid=recid.object_uuid, object_type="rec", pid_type="recid"
```

minor: otherwise you're not using the [(`object_type`, `object_uuid`) index](https://github.com/inveniosoftware/invenio-pidstore/blob/master/invenio_pidstore/models.py#L100), which will become pretty slow once the PIDs table gets big 😅